modules/beta-private-cluster/cluster.tf

```diff
@@ -546,6 +546,15 @@ resource "google_container_cluster" "primary" {
     }
   }
 
+  dynamic "control_plane_endpoints_config" {
+    for_each = var.enable_private_endpoint && var.deploy_using_private_endpoint ? [1] : [0]
+    content {
+      dns_endpoint_config {
+        allow_external_traffic = var.deploy_using_private_endpoint
```

Comment:
@apeabody  Why is this feature gated by `var.enable_private_endpoint && var.deploy_using_private_endpoint` ?

It should de possible to have "DNS-based endpoint" feature enabled without enforcing the usage of the private IP / private endpoint 🤔

Reply:
Hi @legal90 - I was planning to update this behavior as part of a larger change when https://github.com/hashicorp/terraform-provider-google/issues/20369 is released, but I'd be happy to review a PR for just this aspect mow.

Reply:
Hi @apeabody ,
Thank you for your reply! I found your PR, which fixes that behavior and makes it working as it really should (at least, to my view): https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/2313
Kudos! I don't have anything to add there.

P.s. I'm sorry if my previous message sounded unpleasant. Thank you for you hard work on this module! ❤️